test_framework/terraform/harvester/sles/main.tf

```
@@ -0,0 +1,208 @@
+terraform {
+  required_providers {
+    rancher2 = {
+      source  = "rancher/rancher2"
+      version = "~> 4.4.0"
+    }
+  }
+}
+
+provider "rancher2" {
+  api_url   = var.lab_url
+  insecure  = true
+  access_key = var.lab_access_key
+  secret_key = var.lab_secret_key
+}
+
+resource "random_string" "random_suffix" {
+  length           = 8
+  special          = false
+  lower            = true
+  upper            = false
+}
+
+data "rancher2_cluster_v2" "hal-cluster" {
+  name = "hal"
+}
+
+resource "rancher2_cloud_credential" "e2e-credential" {
+  name = "e2e-credential-${random_string.random_suffix.id}"
+  harvester_credential_config {
+    cluster_id = data.rancher2_cluster_v2.hal-cluster.cluster_v1_id
+    cluster_type = "imported"
+    kubeconfig_content = data.rancher2_cluster_v2.hal-cluster.kube_config
+  }
+}
+
+resource "rancher2_machine_config_v2" "e2e-machine-config-controlplane" {
+
+  generate_name = "e2e-machine-config-controlplane-${random_string.random_suffix.id}"
+
+  harvester_config {
+
+    vm_namespace = "longhorn-qa"
+
+    cpu_count = "4"
+    memory_size = "8"
+
+    disk_info = <<EOF
+    {
+        "disks": [{
+            "imageName": "longhorn-qa/image-qn4cg",
+            "size": 100,
+            "bootOrder": 1
+        }]
+    }
+    EOF
+
+    network_info = <<EOF
+    {
+        "interfaces": [{
+            "networkName": "longhorn-qa/vlan104"
+        }]
+    }
+    EOF
+
+    ssh_user = "sles"
+
+    user_data = <<EOF
+#cloud-config
+ssh_authorized_keys:
+  - >-
+    ${file(var.ssh_public_key_file_path)}
+runcmd:
+  - SUSEConnect -r ${var.registration_code}
+  - zypper install -y qemu-guest-agent iptables
+  - - systemctl
+    - enable
+    - '--now'
+    - qemu-guest-agent.service
+EOF
+  }
+}
+
+resource "rancher2_machine_config_v2" "e2e-machine-config-worker" {
+
+  generate_name = "e2e-machine-config-worker-${random_string.random_suffix.id}"
+
+  harvester_config {
+
+    vm_namespace = "longhorn-qa"
+
+    cpu_count = "4"
+    memory_size = "8"
```

```suggestion
    memory_size = "16"
```
I recall @derekbit saying that if we want to use a `v2 volume`, it seems that `16 GB` of host memory is more appropriate.

---

@derekbit Where can we find documentation on this?

---

8 GiB should be okay. Can we try it?

---

> 8 GiB should be okay. Can we try it?

It's been tested.